### PR TITLE
add new platform to compile only to latex

### PR DESCRIPTION
**Pull request by cserteGT3**
As described in #1338 I would like to compile the documentation to latex only. Here's my implmentation for it.
Please let me know what you think about it.

### CODE REVIEW

File: src/Writers/LaTeXWriter.jl [hunk 1]
```diff
@@ -35,13 +35,14 @@ The `makedocs` argument `authors` should also be specified, it will be used for
 
 # Keyword arguments
 
-**`platform`** sets the platform where the tex-file is compiled, either `"native"` (default) or `"docker"`.
+**`platform`** sets the platform where the tex-file is compiled, either `"native"` (default), `"docker"`,
+or "none" which doesn't compile the tex.
```
Comment (mortenpi) [inline src/Writers/LaTeXWriter.jl:39]:
```suggestion
or "none" which doesn't compile the tex.

```

File: docs/src/man/other-formats.md [hunk 2]
```diff
@@ -197,3 +197,16 @@ services:
 ```
 
 to your `.travis.yml` file.
+
+### Compiling to latex only
```
Comment (mortenpi) [inline docs/src/man/other-formats.md:201]:
```suggestion
### Compiling to LaTeX only
```

File: docs/src/man/other-formats.md [hunk 3]
```diff
@@ -197,3 +197,16 @@ services:
...
+
+There's a possibility to save only the `.tex` file and skip the pdf compilation.
```
Comment (mortenpi) [inline docs/src/man/other-formats.md:203]:
```suggestion
There's a possibility to save only the `.tex` file and skip the PDF compilation.
```

File: src/Writers/LaTeXWriter.jl [hunk 4]
```diff
@@ -94,7 +95,7 @@ function mktempdir(args...; kwargs...)
 end
 
 function render(doc::Documents.Document, settings::LaTeX=LaTeX())
-    @info "LaTeXWriter: rendering PDF."
+    @info "LaTeXWriter: rendering tex file."
```
Comment (mortenpi) [inline src/Writers/LaTeXWriter.jl:98]:
```suggestion
    @info "LaTeXWriter: creating the LaTeX file."
```

**Comment by cserteGT3** [comment]
> However, just the `.tex` file is not enough to build the PDF afterwards. The temporary `build/` directory will also contain all the images, the `.sty` file etc. We should probably copy the whole thing over to `build/`?

Agree, I forgot about the other files. Made the changes, to copy the whole source directory to the build directory. Also corrected the documentation, as you requested.

**Comment by inkydragon** [comment]
> We should probably copy the whole thing over to build/?

If I build HTML first and then build LaTeX without PDF, Will I get a mix of HTML and LaTeX file in `build/` dir?

Maybe copy the whole dir to `build/latex/` ?

**Comment by cserteGT3** [comment]
> > We should probably copy the whole thing over to build/?
> 
> If I build HTML first and then build LaTeX without PDF, Will I get a mix of HTML and LaTeX file in `build/` dir?

[`makedocs`](https://juliadocs.github.io/Documenter.jl/stable/lib/public/#Documenter.makedocs) cleans the build directory by default.

> Maybe copy the whole dir to `build/latex/` ?

and `makedocs` has a `build="build"` keyword to specify the build directory. I think that is sufficient.

**Comment by mortenpi** [comment]
> `makedocs` cleans the build directory by default.

That is true. However, I believe currently `platform="none"` will not actually honor `clean = false` currently? The `force=true` in the `cp` will erase the original `build/`?

**Comment by cserteGT3** [comment]
> > `makedocs` cleans the build directory by default.
> 
> That is true. However, I believe currently `platform="none"` will not actually honor `clean = false` currently? The `force=true` in the `cp` will erase the original `build/`?

Yes, I think so. I'll test it later today, and report back.

**Comment by cserteGT3** [comment]
I've tested the followings (on Windows and WSL Linux):

1. `makedocs(..., format = Documenter.HTML(), clean=true)`
1. `makedocs(..., LaTeX(platform = "none"), clean=false)`

The first step is the usual Documenter build, and after the second command nothing is deleted, and the result is a mix of HTML and LaTeX files. Though, I'm not sure why this happens.

> cp(src::AbstractString, dst::AbstractString; force::Bool=false, follow_symlinks::Bool=false)
> 
>  Copy the file, link, or directory from src to dst. force=true will first remove an existing dst.
> 
>  If follow_symlinks=false, and src is a symbolic link, dst will be created as a symbolic link. If follow_symlinks=true and src is a symbolic link, dst will be a copy of the file or directory src refers to. Return dst.

**Comment by mortenpi** [comment]
> and after the second command nothing is deleted, and the result is a mix of HTML and LaTeX files.

I can confirm that, although interestingly when I run `cp("foo", "bar"; force=true)` on some directories, the contents of `bar/` is lost. But Documenter seems to work the way we want it so :man_shrugging:

**Pull request merged**